### ISSUE TITLE
[BUG]-> New-PnPAzureADGroup: Object reference not set to an instance of an object.

### ISSUE THREAD

**Issue by adam13hylo** [pnp/powershell#3363]
### Reporting an Issue or Missing Feature
Issue

### Expected behavior 
When trying to utilise 'New-PnPAzureADGroup' cmdlet I would expect a new AD Group security group to be created, based on my code below: 

```powershell 
New-PnPAzureADGroup -DisplayName "TEST IGNORE" -Description "A description goes here" -IsSecurityEnabled -MailNickname "testignore" -Owners @("myusername@tenant.onmicrosoft.com")
```

### Actual behavior 
When the code above is run or any other variation of 'New-PnPAzureADGroup' as per the documentation, the following error is returned: 

> New-PnPAzureADGroup: Object reference not set to an instance of an object.

### Steps to reproduce behavior

1. Connect to SharePoint admin center authenticating with a certificate (I have also tested this with interactive and device login, however I get the same error):
```powershell 
Connect-PnPOnline -Url "https://tenant-admin.sharepoint.com" -CertificatePath "..\certs\authcert.pfx" -ClientId "0000000.." -Tenant "tenant.onmicrosoft.com"
```
2. Run code to create new Azure AD Group:
```powershell 
New-PnPAzureADGroup -DisplayName "TEST IGNORE" -Description "A description goes here" -IsSecurityEnabled -MailNickname "testignore" -Owners @("myusername@tenant.onmicrosoft.com")
```

### What is the version of the Cmdlet module you are running?
PnP PowerShell Version 2.2.0
PowerShell version 7.3.6

### Which operating system/environment are you running PnP PowerShell on?
- [X] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by veronicageek** [comment]
@adam13hylo - I don't have any issue with `2.2.0` or `2.2.57`. Don't think that'll be the problem, but can you try another method of connecting instead of the certificate and feedback. Thanks.

![image](https://github.com/pnp/powershell/assets/25181757/8e9a4904-3c32-470c-a87a-9490f239e46c)

**Comment by adam13hylo** [comment]
Morning @veronicageek 

Apologies for the delay in getting back to you. 

I have just tried this with both **-DeviceLogin** and also **-Interactive** either of which work: 

![image](https://github.com/pnp/powershell/assets/4320762/ed3b5460-9602-498d-a393-ba6585b6f469)

I have tried this on two tenants just to ensure this isn't a weird permissions issue on one tenant, however, I experience the same issue. I also asked my colleague to test this and I was able to confirm that she also had the same issue when trying to create a new Azure AD group.
